extensions/resteasy/runtime/src/main/java/io/quarkus/resteasy/runtime/standalone/LazyHostSupplier.java

```diff
@@ -20,7 +23,17 @@ public String getRemoteHost() {
         if (initialized) {
             return this.remoteHost;
         } else {
-            SocketAddress socketAddress = request.remoteAddress();
+            SocketAddress socketAddress = null;
+            try {
+                socketAddress = request.remoteAddress();
+            } catch (NullPointerException npe) {
+                // ideally there shouldn't be an exception for this call.
+                // This is just to workaround a current issue in vertx
+                // https://github.com/quarkusio/quarkus/issues/5247
+                // https://github.com/eclipse-vertx/vert.x/issues/3181
+                // TODO: remove this workaround once vertx issue is resolved
+                logger.debug("Ignoring exception that occurred when obtaining remote address of request " + request, npe);
```

Comment:
Logging this at debug level seems a bit low to me, is it really ok considering the remote host will be cached after the NPE occurs?

Reply:
Hello @gwenneg, I decided to use `DEBUG` since the inability to get the client IP shouldn't usually be a concern to the users. In fact the code here (and the previous version of the code) is designed to handle cases where client address isn't available. That's also the reason why caching this null value is fine.

Reply:
Ok, that seems reasonnable. Thanks for the explanation.

Reply:
This has been fixed in Vert.x 3.8.5. So the workaround it not required anymore.